pkg/providers/vsphere/defaults.go

```diff
@@ -101,12 +101,27 @@ func (d *Defaulter) setupDefaultTemplate(ctx context.Context, spec *Spec, machin
 	return nil
 }
 
-func (d *Defaulter) setCloneModeDefaults(ctx context.Context, machineConfig *anywherev1.VSphereMachineConfig, datacenter string) error {
+func max(a, b int) int {
+	if a > b {
+		return a
+	}
+	return b
+}
+
+//nolint:gocyclo // skip cyclomatic complexity check for this function
+func (d *Defaulter) setCloneModeAndDiskSizeDefaults(ctx context.Context, machineConfig *anywherev1.VSphereMachineConfig, datacenter string) error {
 	diskSize, err := d.govc.GetVMDiskSizeInGB(ctx, machineConfig.Spec.Template, datacenter)
```

Comment:
Does this actually return GB or GiB? There is a difference in units if GB so we might need to do some conversion here to do a proper check.

Reply:
this part is a bit ambiguous. The govc command we use to retrieve this info is 
```bash
$ govc device.info -vm bottlerocket-kube-v1-24 -json | grep CapacityInKB
      "CapacityInKB": 23068672,
```
which returns `CapacityInKB`.
vSphere UI also shows the disk size in GB not GiB. It's just CAPV asks for disk size in GiB. But when I create a machine with `25GiB` using CAPV, the vSphere UI and that govc command shows the disk size as `25GB`

Reply:
Oh wow we create a disk with gb instead of gib? Sounds like a bug then right? We can probably create an issue for that then

Reply:
Idk if it's a bug or just ambiguity in the terminology between CAPV and vSphere.
That will require a bit more research so I can create a separate issue to look into that.

Reply:
Create the ticket here: https://github.com/aws/eks-anywhere/issues/4839